Gemfile

```diff
@@ -76,6 +76,10 @@ gem 'shortener'
 # Explicit load to avoid an "unable to load" warning
 gem 'http-2'
 
+# Caching for action responses. Added for slow pages such sample report.
+# See https://github.com/rails/actionpack-action_caching .
+gem 'actionpack-action_caching'
```

Comment:
does it have eviction policies other than the expiration time you set? I'm assuming it has some kind of mechanism that prevents it from filling up too much space? Where are those parameters set?

Reply:
No,  I suppose it could run out of disk space but I doubt that... each report is 20MB max, so we could store 100k in 1TB of disk space. 

When we get to that point I would recommend doing something smarter with redis (LRU).

Reply:
Also, report info should not be 20MB. Probably not even close.

Reply:
I just double checked and the largest cached file is 19M of uncompressed JSON.

Reply:
I don't think it is a good design caching on disk. We need to move this cache to Redis or even S3. We have 6 servers, caching on disk will have a hit cache miss of more than 83%. We will be able to use the cache only 16% of the time.
Not to mention that we will need to evict caches on 6 different machines.
Also, it can lead to inconsistent results, since caches on different machines can potentially present different data.

Reply:
Also, since this data is big, it might worth compressing that before sending to the cache, specially if we use memory caches like redis.

Reply:
why don't we just do memcache?

Reply:
agreed with @davidrissato's point on caching centrally and compression. There's a rails memcache gem that's pretty turn key I think. Why don't we just setup a memcache tier and use that one? https://redislabs.com/lp/rails-memcached/

Reply:
Yea, same comments as above. Does anybody have experience setting up memcache or redis ideally with rails? And do you want to see that change in this PR?

Reply:
Amazon has the elasticcache tier and you can add a docker container for the local use.